### PR TITLE
DEV-3888-test(cameltosnakecase):add test & solutions

### PR DESCRIPTION
**Pull request by Hamzaelkhatri**

### CODE REVIEW

File: solutions/cameltosnakecase.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package solutions
+
+func CamelToSnakeCase(s string) string {
+	if len(s) == 0 {
+		return ""
+	}
+	var result string
+	for i := 0; i < len(s); i++ {
+		if s[i] >= 'A' && s[i] <= 'Z' {
```
Comment (zainabdnaya) [inline solutions/cameltosnakecase.go:9]: The code gives the result different then the subject :
   CamelToSnakeCase("HelloWorld") => _Hello_World
I think u should skip the first character

Comment (Hamzaelkhatri) [reply solutions/cameltosnakecase.go]: fixed

**Review by zainabdnaya — APPROVED** [review]
All Good

**Review by MSilva95 — CHANGES_REQUESTED** [review]
Fix the solution and test accordingly to the comments on the readme for this exercise

**Comment by Hamzaelkhatri** [comment]
@MSilva95 which comments?

**Review by MSilva95 — COMMENTED** [review]
the one in the subject for this exercise.
https://github.com/01-edu/public/pull/1221

File: tests/cameltosnakecase/main.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	student "student"
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+func main() {
+	years := []string{
```
Comment (MSilva95) [inline tests/cameltosnakecase/main.go:11]: add tests to cover this:

- The word does not end on a capitalized letter: (CamelCasE).
- No two capitalized letters shall follow directly each other (CamelCAse).
- Numbers or punctuation are not allowed in the word anywhere (camelCase1).

**Review by MSilva95 — CHANGES_REQUESTED** [review]
add the new test cases and make the solution follow the rules if necessary!

File: tests/cameltosnakecase/main.go (hunk 2 above)
Comment (Hamzaelkhatri) [reply tests/cameltosnakecase/main.go:11]: fixed

**Review by jotapero — APPROVED** [review]
Looks Good :alien:

File: solutions/cameltosnakecase.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package solutions
+
+func containOnlyAlphabe(str string) bool {
```
Comment (tiagocollot) [inline solutions/cameltosnakecase.go:3]: The code seems to be working fine. I would suggest to correct the function name: "containOnlyAlphabet"

File: solutions/cameltosnakecase.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	return true
+}
+
+func isUpper(s rune) bool {
+	return s >= 'A' && s <= 'Z'
+}
+
+func CamelToSnakeCase(s string) string {
+	result := ""
+	if len(s) == 0 || !containOnlyAlphabe(s) {
```
Comment (tiagocollot) [inline solutions/cameltosnakecase.go:18]: Same as above, fix function name typo.

**Review by tiagocollot — CHANGES_REQUESTED** [review]

File: solutions/cameltosnakecase.go (hunk 4 above)
Comment (jotapero) [reply solutions/cameltosnakecase.go]: fixed

**Review by tiagocollot — APPROVED** [review]
LGTM! ⚡

**Pull request merged**